x-pack/plugin/stack/src/main/java/org/elasticsearch/xpack/stack/StackTemplateRegistry.java

```diff
@@ -223,7 +223,20 @@ protected Map<String, ComposableIndexTemplate> getComposableTemplateConfigs() {
     }
 
     private static final List<IngestPipelineConfig> INGEST_PIPELINE_CONFIGS = List.of(
-        new IngestPipelineConfig("logs-default-pipeline", "/logs-default-pipeline.json", REGISTRY_VERSION, TEMPLATE_VERSION_VARIABLE)
+        new IngestPipelineConfig(
+            "logs-json-pipeline",
+            "/logs-json-pipeline.json",
+            REGISTRY_VERSION,
+            TEMPLATE_VERSION_VARIABLE,
+            Collections.emptyList()
+        ),
+        new IngestPipelineConfig(
+            "logs-default-pipeline",
+            "/logs-default-pipeline.json",
+            REGISTRY_VERSION,
+            TEMPLATE_VERSION_VARIABLE,
+            Collections.singletonList("logs-json-pipeline")
```

Comment:
Did you see my [_Note for reviewer_](https://github.com/elastic/elasticsearch/pull/96083#issue-1708957888)?
I think that in this case, as opposed to the simple dependency analysis we do for composable index template, this is not the right thing to do, and will probably make it more brittle. 
If properly tested, I think that adding a pipeline dependency without declaring it here should be noticed immediately.

Reply:
Ha! I managed to completely miss that. I don't know how, I must have just dove right into the changes. 

I'm not sure I agree on automated detection of dependencies making things more brittle. At worst, the brittleness is just as likely to happen. At best, the friction to making changes here is lessened and more time can be spent on making proper tests. 

But like I said above, way out of scope for this pass, and likely for a long time after anyway. Mostly writing the thought out for posterity.

Reply:
The [complexities](https://github.com/elastic/elasticsearch/issues/95522#issuecomment-1543591615) related to doing proper automatic resolution will add their risk and are likely to contain hardly-detected bugs.
In addition, any new or overlooked property that allows the usage of a processor (I only thought of `on_failure`), may break the detection if used with a `pipeline` processor. This can be realized long after this change is made, when someone uses something that is not covered by the algorithm.
Hence the different type of brittleness.

Reply:
Bottom line, thanks for the discussion and the approval 😄